charts/kyverno/templates/validate.yaml

```diff
@@ -1,13 +1,20 @@
-{{- if hasKey .Values "mode" }}
-  {{ fail "mode is not supported anymore, please remove it from your release and use admissionController.replicas instead." }}
-{{- end }}
+{{- if hasKey .Values "mode" -}}
+  {{- fail "mode is not supported anymore, please remove it from your release and use admissionController.replicas instead." -}}
+{{- end -}}
 
-{{- if .Values.admissionController.replicas }}
-  {{- if eq (int .Values.admissionController.replicas) 2 }}
-    {{ fail "Kyverno does not support running with 2 replicas. For a highly-available deployment, select 3 replicas or for standalone select 1 replica." }}
-  {{- end }}
-{{- end }}
+{{- if .Values.admissionController.replicas -}}
+  {{- if eq (int .Values.admissionController.replicas) 2 -}}
+    {{- fail "Kyverno does not support running with 2 replicas. For a highly-available deployment, select 3 replicas or for standalone select 1 replica." -}}
+  {{- end -}}
+{{- end -}}
 
-{{- if eq (include "kyverno.namespace" .) "kube-system" }}
-    {{ fail "Kyverno cannot be installed in namespace kube-system." }}
-{{- end }}
+{{- if eq (include "kyverno.namespace" .) "kube-system" -}}
+  {{- fail "Kyverno cannot be installed in namespace kube-system." -}}
+{{- end -}}
+
+{{- if not .Values.upgrade.fromV2 -}}
+  {{- $v2 := lookup "apps/v1" "Deployment" (include "kyverno.namespace" .) (include "kyverno.fullname" .) -}}
```

Comment:
Minor thing but this will also trigger if there is a deployment which is not owned by helm. Should not really be the case but could catch some people.

Reply:
I'm not sure how to deal with this, tools like argocd for example don't set the helm label IIRC.

Reply:
Not a bad idea if other users have to set this flag. Makes them aware they need to check the new version out.

Reply:
@eddycharly What labels would need to be set and aren't in argocd?  From what I can see you are looking up a resource by name not by labels.

Reply:
@treydock I mean to check if the deployment is owned by helm (`app.kubernetes.io/managed-by: Helm`)